src/controllers/SelectYearController.ts

```diff
@@ -0,0 +1,150 @@
+import Joi from '@hapi/joi';
+import { Session, SessionMiddleware } from 'ch-node-session-handler';
+import { PenaltyList } from 'ch-sdk-node/dist/services/lfp';
+import { Request, Response } from 'express';
+import { UNPROCESSABLE_ENTITY } from 'http-status-codes';
+import { provide } from 'inversify-binding-decorators';
+import { controller } from 'inversify-express-utils';
+import { BaseController, FormActionHandler, FormActionHandlerConstructor } from './BaseController';
+import { SafeNavigationBaseController } from './SafeNavigationBaseController';
+import { FormActionProcessor } from './processors/FormActionProcessor';
+import { FormValidator } from './validators/FormValidator';
+
+import { AuthMiddleware } from 'app/middleware/AuthMiddleware';
+import { loggerInstance } from 'app/middleware/Logger';
+import { PenaltyReferenceRouter } from 'app/middleware/PenaltyReferenceRouter';
+import { Appeal } from 'app/models/Appeal';
+import { ApplicationData, APPLICATION_DATA_KEY } from 'app/models/ApplicationData';
+import { createPenaltyRadioButton } from 'app/models/components/PenaltyRadioButton';
+import { createSchema } from 'app/models/fields/PenaltyChoice.schema';
+import { APPLICATION_DATA_UNDEFINED, SESSION_NOT_FOUND_ERROR } from 'app/utils/CommonErrors';
+import { PENALTY_DETAILS_PAGE_URI, REVIEW_PENALTY_PAGE_URI, SELECT_YEAR_PAGE_URI } from 'app/utils/Paths';
+import { ValidationResult } from 'app/utils/validation/ValidationResult';
+
+const template = 'select-the-year';
+
+const navigation = {
+    previous(): string {
+        return PENALTY_DETAILS_PAGE_URI;
+    },
+    next(): string {
+        return REVIEW_PENALTY_PAGE_URI;
+    },
+    actions: (changeMode: boolean) => {
+        return {
+            continue: changeMode ? 'action=continue&cm=1' : 'action=continue'
+        };
+    }
+};
+
+@provide(Processor)
+class Processor implements FormActionProcessor {
+    public process(request: Request): void | Promise<void> {
+
+        const session = request.session;
+
+        if (!session) {
+            throw SESSION_NOT_FOUND_ERROR;
+        }
+
+        const appData: ApplicationData | undefined = session.getExtraData(APPLICATION_DATA_KEY);
+
+        if(!appData) {
+            throw APPLICATION_DATA_UNDEFINED;
+        }
+
+        const penaltySelected: string = request.body.selectPenalty;
+
+        appData.appeal.penaltyIdentifier.penaltyReference = penaltySelected;
+
+        session.setExtraData(APPLICATION_DATA_KEY, appData);
+    }
+
+}
+
+export const errorMessage: string = 'Select the penalty you want to appeal';
```

Comment:
why not just add this to the `PenaltyChoice.schema.ts` since it is not being re-used elsewhere?